pkg/fieldpath/fieldpath_test.go

```diff
@@ -116,3 +136,83 @@ func TestExtractFieldPathAsString(t *testing.T) {
 		}
 	}
 }
+
+func TestSplitMaybeSubscriptedPath(t *testing.T) {
+	cases := []struct {
+		fieldPath             string
+		expectedPath          string
+		expectedSubscript     string
+		expectedIsSubscripted bool
+	}{
+		{
+			fieldPath:             "metadata.annotations['key']",
+			expectedPath:          "metadata.annotations",
+			expectedSubscript:     "key",
+			expectedIsSubscripted: true,
+		},
+		{
+			fieldPath:             "metadata.annotations['a[b']c']",
+			expectedPath:          "metadata.annotations",
+			expectedSubscript:     "a[b']c",
+			expectedIsSubscripted: true,
+		},
+		{
+			fieldPath:             "metadata.labels['['key']",
+			expectedPath:          "metadata.labels",
+			expectedSubscript:     "['key",
+			expectedIsSubscripted: true,
+		},
+		{
+			fieldPath:             "metadata.labels['key']']",
+			expectedPath:          "metadata.labels",
+			expectedSubscript:     "key']",
+			expectedIsSubscripted: true,
+		},
+		{
+			fieldPath:             "metadata.labels['']",
+			expectedPath:          "metadata.labels['']",
```

Comment:
add a case for `['   ']` ?

Reply:
Done. Also added test for `annotations[ 'foobar' ]`.